### DISCUSSION TITLE
High Memory Usage M2 Max

### DISCUSSION THREAD

**Discussion by Warlock-Merlin** [rustdesk/rustdesk#5988] [General]
I had activity monitor open and I saw that Rust Desk was using a huge amount of memory for what seems to be a light application is this normal behavior? On MacOS 13.5 Mac Studio M2 Max 64gb memory 

![Screenshot 2023-10-11 060521](https://github.com/rustdesk/rustdesk/assets/120424336/f72388f6-8b9d-4ebc-8044-9edcf9e17202)

**Comment by 21pages** [comment]
Is there any ongoing remote connection? If yes, please change the codec to see if there will be significant changes in memory usage.

What version?

**Reply by Warlock-Merlin** [reply]
its RustDesk 1.2.1

How do I change the codec it uses? I am in Ventura 13.5 M2 Max Apple Studio

I am the only remote connection to it from my windows PC x360 HP Spectre.

**Reply by rustdesk** [reply]
Try out 1.2.3 first, it has nothing to do with codec, no codec will use so much memory.

**Reply by Warlock-Merlin** [reply]
Rustdesk is currently using 10 gb of memory, I am connected to the machine for a long period of time doing graphic design work with the mac 50 miles from where I am.

**Reply by rustdesk** [reply]
@21pages  please run 1.2.3 on your mac and play video for hours.

**Reply by Warlock-Merlin** [reply]
I just installed 1.2.3 I will see what happens, I just overwrote the application and choose the aarch64.dmg so hopefully it will be better.

**Reply by Warlock-Merlin** [reply]
When I want to type currently it says "reset by peer" and it doesn't let me type correctly anymore now and then over 3 rustdesk's spawn in my dock to the right.

**Reply by rustdesk** [reply]
Make sure both side upgrade to 1.2.3.

**Reply by Warlock-Merlin** [reply]
Yes I am on 1.2.3 now, with aarch64.dmg application.

**Reply by rustdesk** [reply]
I have been running 1.2.3 with the aarch64.dmg on M1 13.5.1 for hours, no memory leak so far, keyboard working fine also.  It seems a specific issue on your machine/network. We have no clue so far. We will keep watch.

**Reply by Warlock-Merlin** [reply]
I can also video capture too and show you as well :)

**Reply by Warlock-Merlin** [reply]
https://youtu.be/AFh5cf0dzSE

At the end I see what the issue was. The legacy keyboard mode was somehow messing with the connection? maybe my video will help make sense of it.

**Reply by rustdesk** [reply]
@fufesou make sure legacy mode work.

**Reply by rustdesk** [reply]
@Warlock-Merlin what os and rustdesk version on the controlled side?

**Reply by Warlock-Merlin** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/120424336/19b5b294-fda8-4a23-a2ce-6673be9de860)

Windows 11 Pro

**Reply by fufesou** [reply]
@Warlock-Merlin Hi, I cannot reproduce the crash issue.

Can you please run my pre-debug-build? And then sent me the `rustdesk.dump` file.

Please send me an email to let me know your email address. So I can send you the pre-debug-build.

The dump file is located in

```
user process: %userprofile%\AppData\Local\CrashDumps

service process: %SystemRoot%\System32\config\SYSTEMPROFILE\AppData\Local\CrashDumps
```

**Reply by Warlock-Merlin** [reply]
The strangest thing is that legacy mode works again? I just typed this with legacy mode. I even tried 3 different free wifi cafe shops and the problem doesn't happen anymore. it must have been a very extreme edge case cause I can't get it to do it anymore. @fufesou 

The only thing is that from probably me running my laptop 7 - 9 days on end is always that Windows corrupts the longer it runs and that I can get windows 11 to do some pretty crazy stuff when I hibernate the laptop then wake it up then only after doing that 5 times windows just starts to really act haywire with the shutdown not working, start menu doesn't want to open, and the machine is only 7 months old. so yeah windows is a mess as always.

**Reply by fufesou** [reply]
Thanks for your feedback.

I will keep a look at this issue.

If you encounter this issue again, please report it here.

**Reply by Warlock-Merlin** [reply]
yes I pretty much tried all day to recreate it.

**Reply by 21pages** [reply]
After playing videos for several hours with self-monitoring, RustDesk's memory usage consistently stays between 600 and 700MB, on M1, using AV1.

**Reply by Warlock-Merlin** [reply]
All I was doing was just using it for long period of time being connected to it. I am on an M2 Max 64 gb memory Mac Studio Machine. I dont get why M2 hasn't gotten all that great software optimizations out of the gate cause there are some programs where it makes the M1 better than using the M2 Chip when the M2 came out you would think the M2 is just an extension and a refinement of M1 and things would be more common and just better out of the gate. I have watched youtube bench marks on it M1 vs M2 and its really strange imo.

going from 1.2.1 to 1.2.3 did help the massive memory usage go down to only 2 processes at 64 mb each from what is label'ed "rustdesk" I dont know of any dependencies attached to rustdesk or anything.

**Reply by Warlock-Merlin** [reply]
![Screenshot 2023-10-15 222912](https://github.com/rustdesk/rustdesk/assets/120424336/c7b08796-c2b1-4248-ad25-1a24ac7df787)

1.2.3 is still using over 4gb of memory on MacOS 13.5

**Reply by Warlock-Merlin** [reply]
heres more information?? hope it helps.

![Screenshot 2023-10-15 223549](https://github.com/rustdesk/rustdesk/assets/120424336/76bacc37-c9e2-4d33-a36a-8a670ee02787)

![Screenshot 2023-10-15 223611](https://github.com/rustdesk/rustdesk/assets/120424336/f655708f-32ae-4097-9922-9a8b36f8071d)

![Screenshot 2023-10-15 223601](https://github.com/rustdesk/rustdesk/assets/120424336/cca129fd-a713-4a88-a086-3ade88ad067c)

[Sample of RustDesk.txt](https://github.com/rustdesk/rustdesk/files/12911523/Sample.of.RustDesk.txt)

**Comment by rustdesk** [comment]
Maybe related， https://github.com/rustdesk/rustdesk/pull/6461
